### PR TITLE
Refactor docker images in EVerest

### PR DESCRIPTION
**Pull request by andistorm**
## Changes

* Add `build-env-everest` image based on `build-env-base` image
* Fix default values in `deploy-everest-doc.yml` workflow
* Add new input `runner` to `deploy-everest-doc.yml` workflow to allow specifying the runner for the workflow
* Add new job `build-and-push-build-env` to `deploy-everest-doc.yml` workflow to build and push the `build-env-everest` image
* Use `build-env-everest` image in `deploy-everest-doc.yml` workflow

## Related

* https://github.com/EVerest/EVerest/issues/178

## Requires

* https://github.com/EVerest/everest-ci/pull/22

### CODE REVIEW

File: .github/workflows/deploy-everest-doc.yml [hunk 1]
```diff
@@ -35,21 +35,56 @@ on:
...
+      runner:
+        description: 'Which runner to use'
+        default: 'ubuntu-22.04'
+        type: choice
+        options:
+          - 'ubuntu-22.04'
+          - 'large-ubuntu-22.04-xxl'
+env:
+  DOCKER_REGISTRY: ghcr.io
+  EVEREST_CI_VERSION: refactor-docker-images # LTODO: Update this
```
Comment (andistorm) [inline .github/workflows/deploy-everest-doc.yml]:
```suggestion
  EVEREST_CI_VERSION: v1.3.0
```

File: .github/workflows/deploy-everest-doc.yml [hunk 2]
```diff
@@ -35,21 +35,56 @@ on:
...
+    outputs:
+      docker_registry: ${{ env.DOCKER_REGISTRY }}
+      everest_ci_version: ${{ env.EVEREST_CI_VERSION }}
+    steps:
+      - id: check
+        run: |
+          echo "Setting up environment"
+  build-and-push-build-env:
+    name: Build and Push Build Environment
+    uses: everest/everest-ci/.github/workflows/deploy-single-docker-image.yml@refactor-docker-images # LTODO: Update this
```
Comment (andistorm) [inline .github/workflows/deploy-everest-doc.yml]:
```suggestion
    uses: everest/everest-ci/.github/workflows/deploy-single-docker-image.yml@v1.3.0
```

**Review by andistorm — COMMENTED** [review]
TODO before merge

File: .github/workflows/deploy-everest-doc.yml [hunk 3]
```diff
@@ -35,21 +35,56 @@ on:
...
+      runner:
+        description: 'Which runner to use'
+        default: 'ubuntu-22.04'
+        type: choice
+        options:
+          - 'ubuntu-22.04'
+          - 'large-ubuntu-22.04-xxl'
+env:
+  DOCKER_REGISTRY: ghcr.io
+  EVEREST_CI_VERSION: v1.3.0
```
Comment (andistorm) [inline .github/workflows/deploy-everest-doc.yml]:
```suggestion
  EVEREST_CI_VERSION: v1.3.1
```

File: .github/workflows/deploy-everest-doc.yml [hunk 4]
```diff
@@ -35,21 +35,56 @@ on:
...
+    outputs:
+      docker_registry: ${{ env.DOCKER_REGISTRY }}
+      everest_ci_version: ${{ env.EVEREST_CI_VERSION }}
+    steps:
+      - id: check
+        run: |
+          echo "Setting up environment"
+  build-and-push-build-env:
+    name: Build and Push Build Environment
+    uses: everest/everest-ci/.github/workflows/deploy-single-docker-image.yml@v1.3.0
```
Comment (andistorm) [inline .github/workflows/deploy-everest-doc.yml]:
```suggestion
    uses: everest/everest-ci/.github/workflows/deploy-single-docker-image.yml@v1.3.1
```

**Review by andistorm — COMMENTED** [review]
Update everes-ci to `v1.3.1`

**Review by hikinggrass — APPROVED** [review]

**Pull request merged**